packages/steps/src/cacheUtils.ts

```diff
@@ -0,0 +1,15 @@
+import { BuildStepContext } from './BuildStepContext.js';
+
+export interface Cache {
+  disabled: boolean;
+  clear: boolean;
+  key?: string;
+  cacheDefaultPaths?: boolean;
+  customPaths?: string[];
+  paths: string[];
+}
+
+export interface DynamicCacheManager {
+  saveCache(ctx: BuildStepContext, cache: Cache): Promise<boolean>;
+  restoreCache(ctx: BuildStepContext, cache: Cache): Promise<boolean>;
```

Comment:
What does `boolean` mean in this context? Could we use `<void>` and throw-and-catch on error?